mempool/config.go

```diff
@@ -0,0 +1,134 @@
+package mempool
+
+import (
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type (
+	// BidInfo defines the information about a bid.
+	BidInfo struct {
+		Bidder       sdk.AccAddress
+		Bid          sdk.Coin
+		Transactions [][]byte
+	}
+
+	// Config defines the configuration for processing auction transactions. It is
+	// a wrapper around all of the functionality that each application chain must implement
+	// in order for auction processing to work.
+	Config interface {
+		// IsAuctionTx defines a function that returns true iff a transaction is an
+		// auction bid transaction.
+		IsAuctionTx(tx sdk.Tx) (bool, error)
+
+		// GetTransactionSigners defines a function that returns the signers of a
+		// bundle transaction i.e. transaction that was included in the auction transaction's bundle.
+		GetTransactionSigners(tx []byte) (map[string]bool, error)
```

Comment:
Curious. Why does this need to return a map? Do we need constant-time lookup access upstream? If so, let's make this `map[string]struct{}` to avoid unnecessary allocations.